src/status_im/signals/core.cljs

```diff
@@ -56,36 +55,48 @@
                 :peers-count (count (:peers peer-stats)))}))
 
 (rf/defn handle-local-pairing-signals
-  [{:keys [db] :as cofx} event]
+  [{:keys [db] :as cofx} {:keys [type action data] :as event}]
   (log/info "local pairing signal received"
             {:event event})
-  (let [connection-success?             (and (= (:type event)
+  (let [role                            (get-in db [:syncing :role])
+        receiver?                       (= role constants/local-pairing-role-receiver)
+        sender?                         (= role constants/local-pairing-role-sender)
+        connection-success?             (and (= type
                                                 constants/local-pairing-event-connection-success)
-                                             (= (:action event)
+                                             (= action
                                                 constants/local-pairing-action-connect))
-        error-on-pairing?               (contains? constants/local-pairing-event-errors (:type event))
-        completed-pairing?              (and (= (:type event)
+        error-on-pairing?               (contains? constants/local-pairing-event-errors type)
+        completed-pairing?              (and (= type
                                                 constants/local-pairing-event-transfer-success)
-                                             (= (:action event)
+                                             (= action
                                                 constants/local-pairing-action-pairing-installation))
-        logged-in?                      (multiaccounts.model/logged-in? db)
-        ;; since `connection-success` event is received on both sender and receiver devices
-        ;; we check the `logged-in?` status to identify the receiver and take the user to next screen
-        navigate-to-syncing-devices?    (and connection-success? (not logged-in?))
+        received-account?               (and (= type
+                                                constants/local-pairing-event-received-account)
+                                             (= action
+                                                constants/local-pairing-action-pairing-account)
+                                             (and (contains? data :account) (contains? data :password)))
```

Comment:
can we use parameter destructuring here?

Reply:
Updated